### PR TITLE
helper-cli: Add a command to list the processed copyrights

### PR DESCRIPTION
**Pull request by fviernau**

### CODE REVIEW

File: helper-cli/src/main/kotlin/common/Utils.kt [hunk 1]
```diff
@@ -192,6 +193,30 @@ internal fun OrtResult.fetchScannedSources(id: Identifier): File {
...
+            }
+
+            copyRightForPackage
+                .getOrPut(licenseFindings.license, { mutableSetOf() })
+                .addAll(statements)
+        }
+    }
+
+    return result
+}
```
Comment (mnonnenmacher) [inline helper-cli/src/main/kotlin/common/Utils.kt]: Would it make sense to also remove copyright garbage in this function?

File: helper-cli/src/main/kotlin/common/Utils.kt [hunk 2]
```diff
@@ -192,6 +193,30 @@ internal fun OrtResult.fetchScannedSources(id: Identifier): File {
...
+ */
+internal fun OrtResult.getProcessedCopyrightStatements(): Map<Identifier, Map<String, Set<String>>> {
+    val result = mutableMapOf<Identifier, MutableMap<String, MutableSet<String>>>()
+    val processor = CopyrightStatementsProcessor()
+
+    collectLicenseFindings(omitExcluded = true).forEach { (id, findings) ->
+        val copyRightForPackage = result.getOrPut(id, { mutableMapOf() })
+        findings.forEach { (licenseFindings, _) ->
+            val statements = processor.process(licenseFindings.copyrights.map { it.statement }).let {
+                it.processedStatements.keys + it.unprocessedStatements
```
[the remark this answers is not in the document]
Comment (fviernau) [reply helper-cli/src/main/kotlin/common/Utils.kt]: Done.

File: helper-cli/src/main/kotlin/common/Utils.kt [hunk 3]
```diff
@@ -192,6 +193,30 @@ internal fun OrtResult.fetchScannedSources(id: Identifier): File {
...
+/**
+ * Return the processed copyright statements for each package or project contained in this [OrtResult] grouped by
+ * the package and license ID.
+ */
+internal fun OrtResult.getProcessedCopyrightStatements(): Map<Identifier, Map<String, Set<String>>> {
+    val result = mutableMapOf<Identifier, MutableMap<String, MutableSet<String>>>()
+    val processor = CopyrightStatementsProcessor()
+
+    collectLicenseFindings(omitExcluded = true).forEach { (id, findings) ->
+        val copyRightForPackage = result.getOrPut(id, { mutableMapOf() })
```
[the remark this answers is not in the document]
Comment (fviernau) [reply helper-cli/src/main/kotlin/common/Utils.kt]: I prefer the `forEach` variant as I find it has better readability. 
But also here I'm not sure if I guessed the `code snip` you had in mind correctly.
If you still want me to change could you please share the code you have in mind here?

File: helper-cli/src/main/kotlin/common/Utils.kt (hunk 1 above)
Comment (fviernau) [reply helper-cli/src/main/kotlin/common/Utils.kt]: That would make sense. I would prefer to do this in a separate commit / PR though if that's fine with you.

File: helper-cli/src/main/kotlin/common/Utils.kt [hunk 4]
```diff
@@ -192,6 +193,30 @@ internal fun OrtResult.fetchScannedSources(id: Identifier): File {
...
+        findings.forEach { (licenseFindings, _) ->
```
[the remark this answers is not in the document]
Comment (fviernau) [reply helper-cli/src/main/kotlin/common/Utils.kt]: Not 100% sure if I understood you correctly. Hope your comment is addressed.

File: helper-cli/src/main/kotlin/common/Utils.kt (hunk 3 above)
[the remark this answers is not in the document]
Comment (fviernau) [reply helper-cli/src/main/kotlin/common/Utils.kt]: I changed it such that it calls put only once per unique id.

[the remark this answers is not in the document]
Comment (mnonnenmacher) [reply helper-cli/src/main/kotlin/common/Utils.kt]: I mainly wanted to remove the unnecessary `getOrPut` which you did.

File: helper-cli/src/main/kotlin/common/Utils.kt (hunk 4 above)
[the remark this answers is not in the document]
Comment (mnonnenmacher) [reply helper-cli/src/main/kotlin/common/Utils.kt]: Yes, it is.

File: helper-cli/src/main/kotlin/common/Utils.kt (hunk 1 above)
Comment (mnonnenmacher) [reply helper-cli/src/main/kotlin/common/Utils.kt]: Sure.

**Pull request merged**